rs/sns/governance/canister/canister.rs

```diff
@@ -385,6 +383,26 @@ async fn get_metrics(request: GetMetricsRequest) -> get_metrics_response::GetMet
     GetMetricsResponse { get_metrics_result }
 }
 
+/// Returns statistics of the SNS
+///
+/// Cannot be called by other canisters. See also: [`get_metrics_replicated`].
```

Comment:
> Cannot be called by other canisters.

X|

Reply:
> Does composite = true mean that this calls other canisters? 

Yes (more generally, we make some async calls).

Composite queries are a kind of methods separate from regular queries and update methods. 

https://internetcomputer.org/docs/references/ic-interface-spec#nomenclature

> For getting metrics, calling other canisters sounds sus.
> Maybe the solution is just to take away (the need for) composite = true

Thus, inside this methid, we need to make async calls to the SNS Ledger.